[develop] Documentation for ledger eco-system

ericpassmore opened the pull request:
## Summary
- updated README.md linking to developer notes
- added CONTRIBUTING.md with high level steps for updating the eos ledger app
- added Developer Notes with overview of tasks and links to online reasources
- added Guide on How to Run tests with Ragger, incl building Ragger from source
- added Guide on setting up and installing Speculos Emulator
- added notes on setting up python virtual env and building out ledger blue toolset

resolves https://github.com/eosnetworkfoundation/ledger-app/issues/42

ericpassmore committed:
docs for building, testing, developing in ledger eco-system

ericpassmore commented:
@xchapron-ledger could you review these documentation updates. Thank you in advance

xchapron-ledger commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,23 @@
+2. Create a Branch for your work with issue number at the end
+3. Make your changes. Don't forget to update app version in `Makefile`
+4. Check your changes by running [`make scan-build`](./README.md#clang-analyzer)
+5. To validate push your changed to `main` branch to test. This runs git actions to verify build
```
Are you sure of this one?

ericpassmore replied:
LOL, no I am not sure. I very much would like a contribution guide, and I wrote down the steps I use. Open to changes. Any specific suggestions?

xchapron-ledger replied:
I would have just deleted the step 5.
So that you have somehow:
- create a branch and make your changes on it.
- don't forget to bump the version
- run test locally: build, build without security warning and functional tests
- push your changes and create a pull request so that: CI tests will be run and reviewer can review.

To me, no one should push to the "`main` branch to test". Tests can be launch locally and while creating the PR.

xchapron-ledger commented on docs/ledger-blue-setup.md:
````diff
@@ -0,0 +1,80 @@
+   - hardware buttons to accept command
+   - Note if you get error `680f`, delete the app and reload it.
+```
+python -m ledgerblue.loadApp --appFlags 0x240 --path "44'/194'" --curve secp256k1 --tlv --targetId 0x33100004 --targetVersion="1.0.4" --delete --fileName bin/app.hex --appName Eos --appVersion 1.4.3 --dataSize $((0x`cat debug/app.map |grep _envram_data | tr -s ' ' | cut -f2 -d' '|cut -f2 -d'x'` - 0x`cat debug/app.map |grep _nvram_data | tr -s ' ' | cut -f2 -d' '|cut -f2 -d'x'`)) `ICONHEX=\`python ...
````
Why not using `make load` which use under the hood this command?
This should help keeping this part updated :)

ericpassmore replied:
<del>What do you think about removing this whole doc? I have never used ledgerblue or vitualenv. </del> Removed this document.

xchapron-ledger commented on docs/running-tests.md:
```diff
@@ -0,0 +1,75 @@
+
+## Setup
+
+General configuration for pytest is under `ledger-app/tests/functional/conftest.py`. Configuration inclues useful information like the `app name`, the `devices`, and the `ragger backends`. 
```
```suggestion
General configuration for pytest is under `ledger-app/tests/functional/conftest.py`. Configuration includes useful information like the `app name`, the `devices`, and the `ragger backends`. 
```

ericpassmore committed:
Update docs/running-tests.md

Co-authored-by: xchapron-ledger <107417453+xchapron-ledger@users.noreply.github.com>

ericpassmore committed:
fixed docs to align with ledger

- updated links to match updated portal docs
- updated Ragger params for version 1.2
- removed ledgerblue docs they were redundant with ledgerHQ docs
- removing ledgerblue also removed back sideload example
- fixed broken links

ericpassmore committed:
docs macosx need ledgerblue

xchapron-ledger commented on README.md:
````diff
@@ -9,36 +9,86 @@ To use the generic wallet refer to `signTransaction.py`, `getPublicKey.py` or Le
+* Checkout the app builder repository
 
+Note the `./full` path on `docker build` differentiate from the previous version now under `legacy`
 ```
-git clone https://github.com/fix/ledger-vagrant
-cd ledger-vagrant
-vagrant up
+git clone https://github.com/LedgerHQ/ledger-app-builder.git
+cd ledger-app-builder
+sudo docker build -t ledger-app-builder:latest ./full
````
I'm not sure this is what should be done.
Indeed user can rebuild the docker image if they wants. But they can also use is directly with:
`sudo docker run --user "$(id -u)":"$(id -g)" --rm -ti -v "$(realpath .):/app" ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder-lite:latest`

ericpassmore replied:
Correct that is an option. I didn't use the docker image. Reasons are below. I will ask the other EOS engineers what they think. 

We try to use software with a known source and the verified git-hub commits fit that. I didn't find any signatures for the docker image. 
```
docker trust inspect --pretty ledger-app-builder-lite:latest
```

In addition, I have only rebuilt the docker image from source. I've never tried the other approach. I would prefer to document the steps that I've practiced and used.

xchapron-ledger replied:
Sure rebuilding the image works, it's a bit longer but less verifiable that's true.

xchapron-ledger commented on README.md:
````diff
@@ -9,36 +9,86 @@ To use the generic wallet refer to `signTransaction.py`, `getPublicKey.py` or Le
+* copy over the files and enter the docker container
+```
+cd ledger-app
+sudo docker run --rm -ti -v "$(realpath .):/app" ledger-app-builder:latest
````
I would have:
- add the --user option
- use the image from ghcr.io/ledgerhq/
- use the `lite` image `ledger-app-builder-lite`

xchapron-ledger replied:
quite the same on all your docker commands below.

xchapron-ledger commented on docs/running-tests.md:
```diff
@@ -0,0 +1,76 @@
+### Run Tests
+
+You can run emulated tests for a specific device or for all devices. Set `--device` to `all` for all devices.
+To see the GUI for the tests you can use `--display=qt`. If you want to run tests in the background use `--display=headless`. The last option is `--display=text` which simulates the UI inside the terminal. 
```
this values are specific to Speculos only. When running the tests in `test/functional` , you only have the possibility to use `--display` or not.
This is because in fact `--display=text` is somehow deprecated, and `--display=headless` is the default with no screen at all.
So for ragger we only exposed the `--display` option without argument.

ericpassmore committed:
updated parameter usage

xchapron-ledger commented:
@ericpassmore Is it ok for you if I merge it or do you want to add some more change?

ericpassmore commented:
> @ericpassmore Is it ok for you if I merge it or do you want to add some more change?

ok to merge. Thank you!
